pkg/spec/spec.go

```diff
@@ -0,0 +1,252 @@
+package spec
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path"
+	"reflect"
+	"slices"
+	"strconv"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+type File struct {
+	Includes []string       `yaml:"includes"`
+	Spec     map[string]any `yaml:"spec"`
+}
+
+func fileExists(dir fs.FS, filepath string) bool {
+	f, err := dir.Open(filepath)
```

Comment:
Maybe a small comment there that the more common `_, err := os.Stat(filepath); return err == nil` will not work because it operates on a fs.FS? `// fs.FS has no Stat() so we actually need to test-open the file` (or something)

Reply:
I have no idea what I was thinking about, it shouldn't be a problem to use `Stat`. It's still very much prone to TOCTOU, but I cannot see how to solve this without having all files open, and I guess it's not a huge deal.